### PR TITLE
feat: new design for hacker dashboard

### PR DESCRIPTION
**Pull request by logan-r**
### Tickets:

- HCK- #884 

### List of changes:

- Implement new design for dashboard status page and confirm email page
- Refactor dashboard page & hacker dashboard to use hooks

### Type of change:

Please delete options that aren't relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [ ] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes (not quite fully responsive yet, doing that as separate issue)
- [x] Referenced all useful info (issues, tasks, etc)

### Screenshots:
Needs to confirm email
image

Ready to apply
image

Applied
image

Accepted
image

Confirmed
image

Withdrawn
image

Checked in
image

Declined
image

Waitlisted
image

Missed application deadline
image

Accepted but missed RSVP deadline
image

### CODE REVIEW

**Comment by logan-r** [comment]
Builds on https://github.com/hackmcgill/dashboard/pull/872 so that should probably be merged first

**Comment by pierreTklein** [comment]
FYI: Buttons on the staff dashboard appear on a lower z-axis relative to the background image. Therefore, the buttons on the dashboard cannot be interacted with.

image

File: src/features/Status/StatusCTAContainer.tsx [hunk 1]
```diff
@@ -60,39 +54,19 @@ class StatusPage extends React.Component<IStatusPageProps, IStatusPageState> {
...
-              />
-              <BackgroundImage
-                right={'0px'}
-                bottom={'0px'}
-                src={Background}
-                imgHeight={'87%'}
-              />
-            </div>
-          ) : (
+      <div style={{ flex: 1, display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center', padding: '6px 30px 96px 30px' }}>
```
Comment (pierreTklein) [inline src/features/Status/StatusCTAContainer.tsx:57]: Consider `<Flex>` instead of manual styling.

File: src/shared/Form/LabelText.tsx [hunk 2]
```diff
@@ -4,17 +4,26 @@ import { REQUIRED_INPUT } from '../../config';
...
 
 interface ILabelTextProps {
+  // Label text
   label: any;
+
+  // Is this a required field
   required?: boolean;
+
+  // Should this field display a * to let user know it's required
+  // (a field can be required, but still have showRequiredLabel set to false)
```
Comment (pierreTklein) [inline src/shared/Form/LabelText.tsx:14]: Why do we want this scenario?

File: src/shared/HOC/withNavbar.tsx [hunk 3]
```diff
@@ -17,10 +17,12 @@ const withNavbar = <P extends {}>(
 ) =>
   class extends React.Component<P> {
     public render() {
-      return [
-        <Navbar key={0} activePage={options.activePage} />,
-        <Component key={1} {...this.props} />,
-      ];
+      return (
+        <div style={{ height: '100%', minHeight: '100vh', display: 'flex', flexDirection: 'column' }}>
```
Comment (pierreTklein) [inline src/shared/HOC/withNavbar.tsx:21]: This could again be achieved with `Flex`, consider using it :)

**Review by pierreTklein — CHANGES_REQUESTED** [review]

File: src/pages/Account/Create.tsx [hunk 4]
```diff
@@ -1,84 +1,61 @@
...
-      <BackgroundImage
-        src={Coders}
-        top={'60px'}
-        right={'0px'}
-        imgHeight={'90%'}
-        position={'fixed' as 'fixed'}
-      />
-    </MediaQuery>
+      <style jsx>{`
+        /* Prefixing styles with SignUp until styled-jsx plugin integrated with babel to avoid global scoped styling conflicts */
```
[the remark this answers is not in the document]
Comment (logan-r) [reply src/pages/Account/Create.tsx]: Fixed

File: src/shared/Form/LabelText.tsx (hunk 2 above)
Comment (logan-r) [reply src/shared/Form/LabelText.tsx]: So we can have required fields that don't show the required star (e.g. if all form fields are understood to be required)

File: src/features/Status/StatusCTA.tsx [hunk 5]
```diff
@@ -27,116 +32,146 @@ enum DetailedState {
...
+      flexDirection={'column'}
     >
-      <Paragraph
-        color={theme.colors.black80}
-        textAlign={'center'}
-        marginBottom={'3rem'}
-      >
+      <Image
+        src={art}
+        imgHeight={artHeight}
```
[the remark this answers is not in the document]
Comment (logan-r) [reply src/features/Status/StatusCTA.tsx]: Done 👍

File: src/features/Status/StatusCTAContainer.tsx (hunk 1 above)
Comment (logan-r) [reply src/features/Status/StatusCTAContainer.tsx]: Done

File: src/shared/HOC/withNavbar.tsx (hunk 3 above)
Comment (logan-r) [reply src/shared/HOC/withNavbar.tsx]: Done

**Review by pierreTklein — APPROVED** [review]
LGTM. Haven't manually tested but I assume this works on mobile too, and there are no regressions

**Pull request merged**